### DISCUSSION TITLE
[Editor] Use timing/effect points for controlling hitsound volumes

### DISCUSSION THREAD

**Discussion by synchromic** [ppy/osu#28760] [Ideas] [answered]
At the moment, the osu!lazer editor only allows changing the hitsound volume on an object-by-object basis. However, I find that I usually select entire ranges of the map at once and change their hitsound volume. It's somewhat rare that I actually need to set an individual object's volume. Also, it is quite annoying to reselect entire ranges if I need to change their volume again.

I suggest moving back to a method similar to stable, where hitsound volumes are controlled by green lines. This makes it very simple to adjust volumes of entire sections of the song at once, without needing to select all the objects. To allow for more customization, this section-wide volume could be used as a baseline, and then individual hitsound volumes could be adjusted on top of that. This would also solve issues people have with the default hitsound volume being too high. I think these changes would make hitsounding in lazer much more smooth overall.

**Comment by peppy** [comment, marked as answer]
We'll be adding better UI/UX for hit sound application, but it won't be moving back to the timing screen.
